server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -0,0 +1,163 @@
+import copy
+
+from django.core.urlresolvers import reverse
+
+
+class BaseSerializer(object):
+    """
+    Base class to be used for creating serializers
+    """
+
+    def __init__(self, instance=None, context=None, multiple=False):
```

Comment:
At this point instance is not really optional, though it becomes optional with the create/update pattern referenced below.

Unfortunately I know of no absolute way to detect iterability that doesn't leave the possibility of false positives. For example, a string is iterable even though you probably wouldn't consider it to be iterable from a serialization standpoint. The cleanest thing is to ask the user of the class to state their intent.